src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -285,6 +285,8 @@ internal static int CombineHashCodes(int h1, int h2, int h3, int h4, int h5, int
     public struct ValueTuple<T1>
         : IEquatable<ValueTuple<T1>>, IStructuralEquatable, IStructuralComparable, IComparable, IComparable<ValueTuple<T1>>, ITupleInternal
     {
+        private static readonly IEqualityComparer<T1> T1Comparer = EqualityComparer<T1>.Default;
```

Comment:
Good catch. My assumption is that, in doubt, it's safer to use the specific type rather than interface.
Looking at @azhmur's [benchmark](https://github.com/dotnet/corefx/issues/18666), it looks like the code he had benchmarked didn't use interface.